avalon/tools/workfiles/app.py

```diff
@@ -186,201 +240,261 @@ def refresh(self):
+        model = view.model()
+        selection_model = view.selectionModel()
+        selection_model.clearSelection()
 
-            # Scroll list so item is visible
-            QtCore.QTimer.singleShot(100, lambda: self.list.scrollToItem(item))
+        # Select the task
+        mode = selection_model.Select | selection_model.Rows
+        for row in range(model.rowCount(QtCore.QModelIndex())):
+            index = model.index(row, 0, QtCore.QModelIndex())
+            name = index.data(QtCore.Qt.DisplayRole)
+            if name == task:
+                selection_model.select(index, mode)
 
-            self.duplicate_button.setEnabled(True)
-        else:
-            self.duplicate_button.setEnabled(False)
+                # Set the currently active index
+                view.setCurrentIndex(index)
 
-        self.list.setMinimumWidth(self.list.sizeHintForColumn(0) + 30)
+    def get_current_task(self):
+        """Return name of task at current index (selected)
 
-    def save_as_maya(self, file_path):
-        from maya import cmds
-        cmds.file(rename=file_path)
-        cmds.file(save=True, type="mayaAscii")
+        Returns:
+            str: Name of the current task.
 
-    def save_as_nuke(self, file_path):
-        import nuke
-        nuke.scriptSaveAs(file_path)
+        """
+        view = self.widgets["view"]
+        index = view.currentIndex()
+        index = index.sibling(index.row(), 0)  # ensure column zero for name
+        return index.data(QtCore.Qt.DisplayRole)
 
-    def save_changes_prompt(self):
-        self._messagebox = QtWidgets.QMessageBox()
-        self._messagebox.setWindowFlags(QtCore.Qt.FramelessWindowHint)
-        self._messagebox.setIcon(self._messagebox.Warning)
-        self._messagebox.setWindowTitle("Unsaved Changes!")
-        self._messagebox.setText(
-            "There are unsaved changes to the current file."
-            "\nDo you want to save the changes?"
-        )
-        self._messagebox.setStandardButtons(
-            self._messagebox.Yes | self._messagebox.No |
-            self._messagebox.Cancel
-        )
-        result = self._messagebox.exec_()
 
-        if result == self._messagebox.Yes:
-            return True
-        elif result == self._messagebox.No:
-            return False
-        else:
-            return None
+class FilesWidget(QtWidgets.QWidget):
+    """A widget displaying files that allows to save"""
+    def __init__(self, parent=None):
+        super(FilesWidget, self).__init__(parent=parent)
 
-    def open(self, filepath):
+        # Setup
+        self._asset = None
+        self._task = None
+        self.root = None
+        self.host = api.registered_host()
+
+        widgets = {
+            "list": QtWidgets.QListWidget(),
+            "duplicate": QtWidgets.QPushButton("Duplicate"),
+            "open": QtWidgets.QPushButton("Open"),
+            "browse": QtWidgets.QPushButton("Browse"),
+            #"currentFile": QtWidgets.QLabel(),
```

Comment:
Currently unused and hidden to simplify the UI. Showing current file name seems redundant since usually the Host would already show it. Or maybe the current file can be more easily show elsewhere as a standard, e.g. like in the window title. Would love to hear thoughts about this.